src/subsai/models/whisper_api_model.py

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+"""
+Whisper API Model
+
+See [openai/whisper](https://platform.openai.com/docs/guides/speech-to-text)
+"""
+
+import os
+import ffmpeg
+from subsai.models.abstract_model import AbstractModel
+from subsai.utils import _load_config
+from openai import OpenAI
+from pysubs2 import SSAFile
+
+def convert_video_to_audio_ffmpeg(video_file, output_ext="mp3"):
+    # Construct the output file name
+    filename, ext = os.path.splitext(video_file)
+    output_file = f"{filename}.{output_ext}"
+    
+    # Execute the ffmpeg conversion
+    (
+        ffmpeg
+        .input(video_file)
+        .output(output_file)
+        .overwrite_output()
+        .run(quiet=True)
+    )
+
+    return output_file
+
+
+class WhisperAPIModel(AbstractModel):
+    model_name = 'openai/whisper'
+    config_schema = {
+            # load model config
+            'model_type': {
+                'type': list,
+                'description': "One of the official model names listed by `whisper.available_models()`, or "
+                               "path to a model checkpoint containing the model dimensions and the model "
+                               "state_dict.",
+                'options': ['whisper-1'],
```

Comment:
By the way, the API does not provide any other options, just `whisper-1` ?

Reply:
yes: https://platform.openai.com/docs/api-reference/audio/createTranscription?lang=node

```model string Required
ID of the model to use. Only whisper-1 (which is powered by our open source Whisper V2 model) is currently available.```